eeps/eep-0066.md

```diff
@@ -0,0 +1,496 @@
+that indicates its *type*, such as «`$I`» in Basic or Perl,
+where «`$`» is the sigil and «`I`» is the variable.
+
+Here we define a Sigil as a prefix (and a suffix) to a string literal
+that indicates how it should be *interpreted*.  The Sigil is
+a *syntactic sugar* that creates some Erlang term.
+
+A Sigil string literal consists of:
+
+1. The Sigil start character «`~`».
+2. The [Sigil Type][], a name that may be empty.
+3. The [String Content][] within [String Delimiters][].
+4. The [Sigil Suffix][], a name character sequence that may be empty.
+
+### Sigil Type
+
+Immediately following the [Sigil][] start character is the Sigil Type
+which is a name composed of a sequence of characters that are allowed
+as the second or later characters in a variable or an atom.
+In short ISO [Latin-1][] letters, digits, «`_`» and «`@`».
+The [Sigil Type][] may be empty.
+
+The Sigil Type defines how the [Sigil][] syntactic sugar
+shall be interpreted.  The suggested Sigil Types are:
+
+* «»: the vanilla (default) [Sigil][].
+
+  Creates an Erlang `unicode:unicode_binary()`.
+  It is a string represented as a UTF-8 encoded binary,
+  equivalent to applying `unicode:characters_to_binary/1`
+  on the [String Content][].  The [String Delimiters][]
+  and escape characters work as they already do for regular strings,
+  triple-quoted strings, or quoted atoms in Erlang.
+
+  So «`~"abc\d"`» is equivalent to «`<<"abc\d"/utf8>>`», and
+  «`~'abc"d'`» is equivalent to «`<<"abc\"d"/utf8>>`».
+
+  Regular strings honour escape sequences but triple-quoted strings
+  are verbatim, so «`~"`» is equivalent to «`~b"`» but
+  «`~"""`» is equivalent to «`~B"""`», as described below.
+
+  A simple way to create strings as UTF-8 binaries is supposedly
+  the first and most desired missing string feature in Erlang.
+  This sigil does just that.
+
+* «`b`»: `unicode:unicode_binary()`
+
+  Creates a UTF-8 encoded binary, handling escape characters
+  in the string content.  Other features such as string interpolation
+  will require another Sigil Type or using the [Sigil Suffix][].
+
+  In Elixir this corresponds to the «`~s`» sigil, a [string][4].
+
+* «`B`»: `unicode:unicode_binary()`, verbatim.
+
+  Creates a UTF-8 encoded binary, with verbatim string content
+  in that only the [end delimiter][] character can be escaped
+  with a «`\`» character.
+
+  In Elixir this corresponds to the «`~S`» sigil, a [string][4].
+
+* «`s`»: `string()`.
+
+  Creates a Unicode codepoint list, handling escape characters
+  in the string content.  Other features such as string interpolation
+  will require another Sigil Type or using the [Sigil Suffix][].
+
+  In Elixir this corresponds to the «`~c`» sigil, a [charlist][5].
+
+* «`S`»: `string()`, verbatim.
+
+  Creates a Unicode codepoint list, with verbatim string content
+  in that only the [end delimiter][] character can be escaped
+  with a «`\`» character.
+
+  In Elixir this corresponds to the «`~C`» sigil, a [charlist][5].
```

Comment:
A string in Erlang, according to the `string()` module, is any `unicode:chardata()`. According to types, `string()` is `[char()]`. For this reason, I would go with `~c` to avoid any source of confusion, but it is not a strong opinion. :)

Reply:
I was thinking about the built in type `string()`, but OK, there is a possible source of confusion here...